src/main/webapp/search-script.js

```diff
@@ -0,0 +1,122 @@
+function collectUnitsFromSearch() {
+  const inputElems = document.getElementsByName("search-units");
+  const addedUnits = []
+  for (let elem of inputElems) {       
+    if (elem.checked){
+      addedUnits.push(elem.value);
+    }
+  }
+  return addedUnits;
+}
+
+function collectUnitsFromInput() {
+  const inputElems = document.getElementsByName("num-units");
+  const addedUnits = []
+  for (let elem of inputElems) {       
+    if (elem.checked){
+      addedUnits.push(elem.value);
+    }
+  }
+  return addedUnits;
+}
+
+function showCourses() {
+  const courseResults = document.getElementById("search-results");
+  const courseName = document.getElementById("search-course").value;
+  const profName = document.getElementById("search-prof").value;
+  const termName = document.getElementById("search-term").value;
+  const units = collectUnitsFromSearch();
+  const school = getUserSchool();
+  courseResults.innerHTML = "";
+  const url = new URL("/search", window.location.origin);
+  url.searchParams.set("courseName", courseName);
+  url.searchParams.set("profName", profName);
+  url.searchParams.set("units", units);
+  url.searchParams.set("term", termName);
+  url.searchParams.set("schoolName", school)
+  fetch(url)
+  .then(response => response.json())
+  .then((searchResults) => {
+    if (searchResults.hasOwnProperty("message")) { // TODO: Make a div for this message so specific styling is easier.
+        courseResults.innerHTML += searchResults.message + "<br />";
```

Comment:
FYI this allows for XSS injection pretty easily, as hackers can change the searchResults.message variable to their own script tag and do whatever they want with host page permissions.

Don't worry about it for now, but if we were making a real website this would be forbidden (and we'd have protected methods for accessing innerHTML/etc).

Reply:
ooh that's so interesting, had no idea. I'll look into that!